libnetwork/cni/network.go

```diff
@@ -295,6 +296,38 @@ func (n *cniNetwork) DefaultInterfaceName() string {
 	return cniDeviceName
 }
 
+// NetworkInfo return the network information about binary path,
+// package version and program version.
+func (n *cniNetwork) NetworkInfo() types.NetworkInfo {
+	path := ""
+	packageVersion := ""
+	for _, p := range n.cniPluginDirs {
+		ver := cutil.PackageVersion(p)
+		if ver != "Unknown" {
+			path = p
+			packageVersion = ver
+			break
+		}
+	}
+
+	info := types.NetworkInfo{
+		Backend: "cni",
+		Package: packageVersion,
+		Path:    path,
+	}
+
+	dnsPath := filepath.Join(path, "dnsname")
+	dnsPackage := cutil.PackageVersion(dnsPath)
+	if _, err := os.Stat(dnsPath); err == nil {
```

Comment:
At least for dnsname you can get the version from the binary as well:
```
$ /usr/libexec/cni/dnsname --version
CNI dnsname plugin
version: 1.3.1
commit: unknown
```

Reply:
Add a wrapper function `ProgramVersionDnsname()` to get the dnsname information.
AFAICS, dnsname --version returns the information to stdErr.

```console
# /usr/libexec/cni/dnsname --version > /dev/null
CNI dnsname plugin
version: 1.3.1
commit: unknown

# /usr/libexec/cni/dnsname > /dev/null
CNI dnsname plugin
version: 1.3.1
commit: unknown


# /usr/libexec/cni/dnsname --version 2> /dev/null

# /usr/libexec/cni/dnsname 2> /dev/null
```